src/components/LayoutTree.tsx

```diff
@@ -438,6 +444,35 @@ const LayoutTree = () => {
     ),
   )
 
+  const hoveringPath = useSelector(state => state.hoveringPath)
+  const contextParentPath = parentOf(hoveringPath || [])
+
+  const isSortedContext = useSelector(state => {
+    const dropTargetId = head(state.hoveringPath || [])
+
+    // Check if the drop target is on sorted context children or on its parent.
+    const isContextChildren = attributeEquals(state, head(contextParentPath), '=sort', 'Alphabetical')
+    const isContextParent = attributeEquals(state, dropTargetId, '=sort', 'Alphabetical')
+
+    return isContextParent || isContextChildren
+  })
+
+  const dragDropManager = useDragDropManager()
+
+  const sourceThought = useSelector(state => {
+    const monitor = dragDropManager.getMonitor()
+    const item = monitor.getItem() as DragThoughtItem
+
+    // Check if the dragged item is a thought and the drop zone is not a subthought
+    const isThought = item?.zone === 'Thoughts' && state.hoverZone === DropThoughtZone.ThoughtDrop
+    const sourceThoughtId = head(item?.path || [])
+
+    const sourceThought = isThought ? getThoughtById(state, sourceThoughtId) : null
+    return sourceThought
+  })
+
+  const newRank = useSelector(state => getSortedRank(state, head(contextParentPath), sourceThought?.value || ''))
+
```

Comment:
Rather than overloading the `<LayoutTree>` file with all of this code, I would prefer to see it moved somewhere else in a separate file. @raineorshine recommended creating a custom hook, which is one of the better options for encapsulating this logic.